### DISCUSSION TITLE
help understanding tags and statusmap

### DISCUSSION THREAD

**Discussion by eostermueller** [flant/grafana-statusmap#160] [Q&A]
Hi, 
I'm sure this isn't a bug, thanks in advance for any help.

First of all, statusmap is very cool -- I'm most of the way to displaying a very nice timeline for our batch process using status map.  One of the main goals is to show where parallel processing is/isn't used.  This is the only grafana plugin I've found that can do that.

However, there there are some extra "features" showing up on my statusmap that I'd like to remove.  Suggestions for how to 'remove' these things? 

1) In the screenshot below, I put a white rectangle around a number of rows I want to go away.  Those are tag names in my influx data, and they're all tags attached to the good data, the rows above my white rectangle.
2) I think once the above is fixed, the message in the white oval ("Data has value with undefined color") will go away.

[dump of all data, about 5k, from influx](https://github.com/flant/grafana-statusmap/files/3135939/data-from-influx.txt)

[json for my panel](https://github.com/flant/grafana-statusmap/files/3135940/myPanelJson.txt)

Here is an example of one data point I'm sending to influx:
`batch_times,batchEnv=FOO_DEV,step=S_04_SERIAL_B_@BATCH,batchEndDate=2019-02-10,S_04_SERIAL_B_@BATCH=400 value=400 1556741570000000000`

![image](https://user-images.githubusercontent.com/175773/57041037-ce776900-6c26-11e9-9be9-ad356384a229.png)

**Comment by diafour** [comment]
Hi! Thank you for the kind words!

1. The best way is to eliminate unused data in query (the datasource layer), not in a panel settings (the view layer). You said that you want get fields and drop tags. May be `*::field` helps you? https://docs.influxdata.com/influxdb/v1.7/query_language/data_exploration/#select-all-fields-from-a-single-measurement

2. "Data has value with undefined color" message is shown if data contain values that not mapped to colors. So your statement is absolutely right.

**Comment by eostermueller** [comment]
Thanks for the quick response.  The "*::field"  didn't work, but I excluded some tags and I'm one step further.
I'm left with one row in my statusmap that is the name of my measurement, which is "batch_times" -- see the screenshot below, the content in the white rectangle is what I'm trying to get rid of.
Any suggestions for getting rid of this?
Here is my query:
`SELECT  value, /S_/ FROM "batch_times" where "batchEndDate" = '$BATCH_END_DATE' and "batchEnv" = '$BATCH_ENV' `
The partial influxdb response is below.
--Erik
![image](https://user-images.githubusercontent.com/175773/57085196-514d0200-6cc1-11e9-8255-7b2f24876b22.png)


`{
    "results": [
        {
            "statement_id": 0,
            "series": [
                {
                    "name": "batch_times",
                    "columns": [
                        "time",
                        "value",
                        "S_01_FIRST_STEP_@BULK",
                        "S_01_SECOND_STEP_@BULK",
                        "S_02_PARALLEL_A_@ETL",
                        "S_02_PARALLEL_B_@ETL",
                        "S_03_SERIAL_A_@BATCH",
                        "S_04_SERIAL_B_@BATCH",
                        "S_05_SERIAL_C_@BATCH"
                    ],
                    "values": [
                        [
                            1556741543000,
                            200,
                            "200",
                            null,
                            null,
                            null,
                            null,
                            null,
                            null
                        ],
`

**Comment by diafour** [comment]
As i can understand, `batch_times` row is a `value` in Select statement. Can you drop it?

**Comment by eostermueller** [comment]
I will work towards that, thanks.
